app/presenters/hyrax/file_set_presenter.rb

```diff
@@ -100,9 +112,15 @@ def link_presenter_class
     def fetch_parent_presenter
       ids = Hyrax::SolrService.query("{!field f=member_ids_ssim}#{id}", fl: Hyrax.config.id_field)
                               .map { |x| x.fetch(Hyrax.config.id_field) }
-      Hyrax::PresenterFactory.build_for(ids: ids,
-                                        presenter_class: WorkShowPresenter,
-                                        presenter_args: current_ability).first
+      ids.each do |id|
+        doc = ::SolrDocument.find(id)
+        next if current_ability.can?(:edit, doc)
+        raise WorkflowAuthorizationException if doc.suppressed? && current_ability.can?(:read, doc)
+      end
+      rv = Hyrax::PresenterFactory.build_for(ids: ids,
```

Comment:
is this variable assignment doing something?

Reply:
I'm removing that local variable in a later commit.